### DISCUSSION TITLE
Add search fields with  `user__username` for `TokenAdmin`

### DISCUSSION THREAD

**Discussion by rnv812** [encode/django-rest-framework#8927] [Ideas & Suggestions] [answered]
I would like to search created tokens by username. I understand that I can inherit from TokenAdmin and create custom TokenAdmin, but I think it would be great if it is included in default functionality.
![Screenshot_20230403_151329](https://user-images.githubusercontent.com/63803175/229505778-20f6d427-3afc-4f93-b5a2-440ea6c26f74.png)

**Comment by auvipy** [comment]
since it is a minor usability improvement, I would consider a contribution in this regard as a positive one.

**Reply by rnv812** [reply]
Should I create issue from this discussion?

**Reply by auvipy** [reply, marked as answer]
you can come with a PR

**Reply by rnv812** [reply]
Here it is #8934

**Comment by hoenie-ams** [comment]
Just FYI: #8934 actually broke my code, because I am using `django-allauth` without `username` (but `email` instead).

**Reply by rnv812** [reply]
As one of the solutions to your problem you can use method described here: https://github.com/pennersr/django-allauth/issues/723#issuecomment-56639178

**Reply by rnv812** [reply]
But actually it's seem to me that no search fields should be initially set for better compatibility with third party libraries. So probably we should revoke this PR @auvipy

**Reply by hoenie-ams** [reply]
Yeah I fixed it again by changing it back in 'TokenAdmin'. Just wanted to let you know. Thanks for all your work on this!
